### PR TITLE
Rebase #168: Connection#unescape handles numerics, #search guards against nil queued_reads

### PR DESCRIPTION
**Pull request by jch**
See #168 for background and feedback.

### CODE REVIEW

**Comment by jch** [comment]
@mtodd I think I've addressed all of your feedback in #168, could you look over this one more time?

File: lib/net/ldap/connection.rb [hunk 1]
```diff
@@ -571,12 +571,13 @@ def search(args = nil)
...
     end # instrument
   ensure
+
     # clean up message queue for this search
     messages = message_queue.delete(message_id)
 
     # in the exceptional case some messages were *not* consumed from the queue,
     # instrument the event but do not fail.
-    unless messages.empty?
+    if !messages.nil? && !messages.empty?
```
Comment (mtodd) [inline lib/net/ldap/connection.rb]: :ok_hand:

File: lib/net/ldap/filter.rb [hunk 2]
```diff
@@ -645,9 +645,8 @@ def match(entry)
 
   ##
   # Converts escaped characters (e.g., "\\28") to unescaped characters
-  # ("(").
   def unescape(right)
-    right.gsub(/\\([a-fA-F\d]{2})/) { [$1.hex].pack("U") }
+    right.to_s.gsub(/\\([a-fA-F\d]{2})/) { [$1.hex].pack("U") }
```
Comment (mtodd) [inline lib/net/ldap/filter.rb]: :+1:

**Comment by mtodd** [comment]
Looks good! :shipit:

**Pull request merged**